### ISSUE TITLE
the hint field value is not UTF-8 encoded

### ISSUE THREAD

**Issue by George-br** [cartertemm/AI-content-describer#27]
Hello!

if you write a prompt in the application settings in a language other than English, the settings cannot be saved

**Comment by cartertemm** [comment]
Is this on the latest version? I tried using the following prompt, but the settings are saving without an issue, and I don't see anything in the log:

"描述这张图片的细节尽可能详细。您的回答应该用中文写成。"

**Comment by George-br** [comment]
> Це в останній версії? Я спробував скористатися таким запитом, але налаштування зберігаються без проблем, і я нічого не бачу в журналі:
> 
> "描述这张图片的细节尽可能详细。您的回答应该用中文写成。."

Yes, it is in the latest version, if you write the text of the prompt in Ukrainian, it is impossible to save the settings
Here is an example in Ukrainian, you can copy and paste it into the hint field and try to save it.

«Опишіть цю картинку якомога детальніше. Ваша відповідь має бути написана українською мовою».

**Comment by George-br** [comment]
> Is this on the latest version? I tried using the following prompt, but the settings are saving without an issue, and I don't see anything in the log:
> 
> "描述这张图片的细节尽可能详细。您的回答应该用中文写成。"

Yes, it is in the latest version, if you write the text of the prompt in Ukrainian, it is impossible to save the settings
Here is an example in Ukrainian, you can copy and paste it into the hint field and try to save it.
«Опишіть цю картинку якомога детальніше. Ваша відповідь має бути написана українською мовою».

**Comment by daniel-luper** [comment]
Hi George, sorry you've had trouble updating the prompt!

I was able to save your Ukrainian prompt in the settings for Claude-3-Haiku without any problems. 
Could you provide some more context around your bug? 

Also, did you press the "Apply" or "Ok" buttons after updating the prompt?

**Comment by cartertemm** [comment]
Hello @George-br,

I just wanted to follow up on this. Are you still experiencing the same behavior after building the addon from main? I've tried on multiple machines without fail.

If so, can you tell me what you have set for the default Windows locale and language?

Thanks!

**Comment by George-br** [comment]
@cartertemm
Hello!
Now the hint in the field is saved, everything is fine.
But after clicking the OK or Apply button, the settings dialog does not close automatically, as it should, in general, if you change any settings in the application after clicking OK, nothing happens with the settings dialog, although the settings are applied 
you need to press the ESC key to close the settings dialog.

**Comment by schulle4u** [comment]
I'm facing the same issue with add-on version 2025.06.05. As soon as I switch to a German prompt in any of the models, neither the model configuration nor the main NVDA settings dialogue will close after clicking OK. Removing all German umlauts from the prompt solves the problem.

```
Beschreibe dieses Bild kurz und prägnant, aber so detailliert wie möglich. Wenn Text vorhanden ist, gebe diesen bitte genau so wieder, wie er abgebildet ist.
```

**Comment by cartertemm** [comment]
Hello @schulle4u,

Is your windows locale by chance set to German?

With the Windows and NVDA default languages set to English, I was able to use this prompt, and upon pressing OK I was able to describe an image without difficulty.

As soon as I can reproduce the problem on my end I will start working on a fix, I just need to isolate the cause.

**Comment by schulle4u** [comment]
@cartertemm yes, the system locale is set to german. ('de_DE', 'cp1252')

**Comment by schulle4u** [comment]
Thanks, works perfectly for me now.
